desc/protoparse/lexer.go

```diff
@@ -287,14 +288,21 @@ func (l *protoLex) Lex(lval *protoSymType) int {
 				return _FLOAT_LIT
 			}
 			// integer! (decimal or octal)
-			ui, err := strconv.ParseUint(token, 0, 64)
```

Comment:
By virtue of using this with base 0, this was inadvertently adding support for a few things that `protoc` does not actually support:
1. a prefix of `0b` to indicate a binary (base 2) literal
2. a prefix of `0o` to indicate octal (in addition to just `0` prefix)
3. arbitrary underscores, which are ignored and allowed just for readability of large constants (e.g. `1_000_000`).

So this change explicitly specifies a base of 8 or 10 to avoid the above support. This variance was accidentally introduced when numeric parsing was simplified in #414. That change made `readNumber` much more lenient, and add simpler validity checks later. But with `readNumber` more lenient, it began accepting tokens like the above (e.g. allows `_` and `0b` and `0o` prefixes), which interact incorrectly with the use of `strconv.ParseUint` with base=0.